packages/ui-components/src/components/NatPurseSelector.jsx

```diff
@@ -0,0 +1,85 @@
+// import React from 'react';
+// import { MenuItem, TextField } from '@material-ui/core';
+import clsx from 'clsx';
+// import { makeStyles } from '@material-ui/core/styles';
+import { MathKind } from '@agoric/ertp';
+
+import makeNatPurseListItemDetailed from './NatPurseListItemDetailed';
+
+const makeNatPurseSelector = (
+  React,
+  MenuItem,
+  TextField,
+  ListItemIcon,
+  ListItemText,
+  PurseIcon,
+  makeStyles,
+) => ({
+  purses: pursesUnfiltered,
+  purseSelected,
+  onChange,
+  disabled,
+  error,
+}) => {
+  const ListItem = makeNatPurseListItemDetailed(
+    React,
+    MenuItem,
+    ListItemIcon,
+    ListItemText,
+    PurseIcon,
+    makeStyles,
+  );
+
+  const useStyles = makeStyles(() => ({
+    select: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    noPadding: {
+      paddingTop: 0,
+      paddingBottom: 0,
+    },
+  }));
+
+  const classes = useStyles();
+
+  const noPurses = (
+    <MenuItem key={null} value={null}>
+      No purses
+    </MenuItem>
+  );
+
+  let items = noPurses;
+  if (pursesUnfiltered === null) {
+    pursesUnfiltered = [];
+  }
+
+  const isNatPurse = ({ displayInfo: { amountMathKind } }) =>
+    amountMathKind === MathKind.NAT;
+
+  const purses = pursesUnfiltered.filter(isNatPurse);
```

Comment:
we have a lot of uses of filtering purses for a particular issuer. sometimes non-fungible included (e.g., show me the baseball card purses).  This seems like the place to add those filter options?

Reply:
Good idea! I added an option for that.